02-documentation/developer-guides/building-a-nextjs-blog-with-squidex.md

```diff
@@ -0,0 +1,588 @@
+  return {
+    props: {
+      posts: data.queryPostsContents,
+    },
+  };
+}
+```
+
+This sends a GraphQL query to your Squidex app to get the slug and titles of all your blog posts and it passes this data to your page.
+
+In Next.js, `getStaticProps` is a function that is run at build time to fetch and store data needed to render a page. To learn more about fetching data in Next.js see [Data Fetching in Next.js](https://nextjs.org/docs/basic-features/data-fetching). Save the file and refresh your browser to fetch page data from the CMS.
+
+To make this data available to the `Home` page component of `index.js`, pass posts as a prop to `Home` in index.js:
+
+```js
+export default function Home({ posts }) {
+...
+}
+```
+
+Although the title and slugs of your blog posts are now available to the browser, you cannot see them yet. You will now create a `BlogItem` component that will be used to display and link to all your posts.
+
+Create a `blogItem.js` file in the `squidex-blog/components` folder with the following contents:
+
+```jsx
+import Link from "next/link";
+
+export default function BlogItem({ title, slug }) {
+  return (
+    <div className="blog-item">
+      <Link href={slug}>
+        <a className="blog-item-title">{title}</a>
+      </Link>
+    </div>
+  );
+}
+```
+
+To use `BlogItem` on your homepage, import it in `pages/index.js`
+
+```jsx
+import BlogItem from "../components/blogItem";
+```
+
+And in the `Home` function in `index.js`, after the `Head` tag, add the following:
+
+```jsx
+<div className="blog-post-list">
+  {posts.map((post) => (
+    <BlogItem
+      title={post.flatData.title}
+      slug={post.flatData.slug}
+      key={post.id}
+    />
+  ))}
+</div>
+```
+
+This loops through your posts and creates a link to each of them.
+
+Your completed `index.js` file will look like this:
+
+```jsx
+import Head from "next/head";
+import fetchAPI from "../lib/squidex";
+import BlogItem from "../components/blogItem";
+
+export default function Home({ posts }) {
+  return (
+    <div>
+      <Head>
+        <title>My blog</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <div className="blog-post-list">
+        {posts.map((post) => (
+          <BlogItem
+            title={post.flatData.title}
+            slug={post.flatData.slug}
+            key={post.id}
+          />
+        ))}
+      </div>
+    </div>
+  );
+}
+
+export async function getStaticProps() {
+  const data = await fetchAPI(`
+  {
+    queryPostsContents {
+      id,
+      flatData {
+        title,
+        slug,
+      }
+    }
+  }
+  `);
+  return {
+    props: {
+      posts: data.queryPostsContents,
+    },
+  };
+}
+```
+
+If you look at your blog homepage in a browser, you will see that the titles of your posts are shown and they link to the page where the article can be read.
+
+Now that you have successfully created links to your posts on the homepage, you will create the pages where your blog posts can be read.
```

Comment:
Who can read it?

Perhaps "...you will create page with a list of blog posts and a page to show one blog post."?